pkg/generator/certificate_reconciler.go

```diff
@@ -116,7 +116,15 @@ func (r *CertificateReconciler) createSecret(ctx context.Context, params certPar
 
 	newSecret := secret.AsSecret()
 
-	GenerateInputs{params}.Add(newSecret.Annotations)
+	if newSecret.Annotations == nil {
+		newSecret.Annotations = map[string]string{
+			"secretgen.k14s.io/generate-inputs": "",
```

Comment:
thanks! this PR is almost ready.  i think we just need to remove this line. here's a little reasoning.: 

1.) we're about to call the `Add` method, and it will set the annotation. No need to do it in advance.
2.) The annotation's key name here is hardcoded straight into the string, but it's actually a constant in the other file, so we don't need to leak that information or duplicate it here. 

so anyway please just make an empty map `{}` and that should take care of it.